clang/lib/Serialization/ASTWriterDecl.cpp

```diff
@@ -798,6 +798,17 @@ void ASTDeclWriter::VisitFunctionDecl(FunctionDecl *D) {
     }
   }
 
+  if (D->getFriendObjectKind()) {
```

Comment:
As stated in https://github.com/llvm/llvm-project/issues/122493#issuecomment-2602473058, this is intended. It would be helpful to include comments explaining why this logic is applied exclusively to friend objects.